app/src/org/commcare/android/util/TemplatePrinterUtils.java

```diff
@@ -0,0 +1,184 @@
+     * http://stackoverflow.com/questions/2206378/how-to-split-a-string-but-also-keep-the-delimiters
+     *
+     * @param input String to split
+     * @param delimiterStart Start delimiter; will split immediately before this delimiter
+     * @param delimiterEnd End delimiter; will split immediately after this delimiter
+     * @return Split string array
+     */
+    public static String[] splitKeepDelimiter(
+            String input,
+            String delimiterStart,
+            String delimiterEnd) {
+
+        String delimiter = String.format(FORMAT_REGEX_WITH_DELIMITER, delimiterStart, delimiterEnd);
+
+        return input.split(delimiter);
+    }
+
+    /**
+     * @param file the input file
+     * @return A string representation of the entire contents of the file
+     * @throws IOException
+     */
+    public static String docToString(File file) throws IOException {
+        StringBuilder builder = new StringBuilder();
+        BufferedReader in = new BufferedReader(new FileReader(file));
+        String str;
+        while ((str = in.readLine()) != null) {
+            builder.append(str);
+        }
+        in.close();
+        return builder.toString();
+    }
+
+    /**
+     * Writes the given string, encrypted, to the file location specified
+     *
+     * @param fileText the content to write
+     * @param outputPath the path of the file to write to
+     * @throws IOException
+     */
+    public static void writeStringToFile(String fileText, String outputPath) throws IOException {
+        try {
+            Cipher encrypter = Cipher.getInstance("AES");
+            encrypter.init(Cipher.ENCRYPT_MODE, KEY);
+            FileOutputStream fos = new FileOutputStream(new File(outputPath));
+            CipherOutputStream cos = new CipherOutputStream(fos, encrypter);
+            BufferedWriter out = new BufferedWriter(new OutputStreamWriter(cos));
+            out.write(fileText);
+            out.close();
+        } catch (NoSuchAlgorithmException e) {
+            e.printStackTrace();
+            throw new RuntimeException(e);
+        } catch (NoSuchPaddingException e) {
+            e.printStackTrace();
+            throw new RuntimeException(e);
+        } catch (InvalidKeyException e) {
+            e.printStackTrace();
+            throw new RuntimeException(e);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Reads back from the encrypted file generated by the above and returns a string
+     * representation of the file's contents
+     *
+     * @param readFromPath the path to the file from which to read
+     * @return a string representation of the file's contents
+     */
+    public static String readStringFromFile(String readFromPath) throws IOException {
+        try {
+            Cipher decrypter = Cipher.getInstance("AES");
+            decrypter.init(Cipher.DECRYPT_MODE, KEY);
```

Comment:
Hm yea kinda figured as I was doing it that I must be duplicating existing stuff, but also needed this to do something pretty specific in terms of the cipher stream. Happy to do a refactor that puts common stuff into a util class, but would definitely want that to be its own PR.